### DISCUSSION TITLE
is 16.4.6 working for you?

### DISCUSSION THREAD

**Discussion by CoderCharlie** [invertase/react-native-firebase#6769] [General]
if so, do you mind sharing versions of react-native, expo(if used), and your Podfile?
I am using react-native-firebase 16.4.6, react-native 0.70.6, expo 47.0.8, and got constant crash at startup. 

```
pod_folder = Pathname.new(__FILE__).dirname.realpath

require File.join(File.dirname(`cd #{pod_folder} && node --print "require.resolve('expo/package.json')"`), "scripts/autolinking")
require File.join(File.dirname(`cd #{pod_folder} && node --print "require.resolve('react-native/package.json')"`), "scripts/react_native_pods")
require File.join(File.dirname(`cd #{pod_folder} && node --print "require.resolve('@react-native-community/cli-platform-ios/package.json')"`), "native_modules")

platform :ios, '13.0'
$FirebaseSDKVersion = '10.2.0' # https://rnfirebase.io/

install! 'cocoapods', :deterministic_uuids => false

production = ENV["PRODUCTION"] == "1"

target 'TestApp' do
  use_expo_modules!
  use_frameworks! :linkage => :static # use frameworks with static for hermes https://rnfirebase.io/
  
  post_integrate do |installer|
    begin
      expo_patch_react_imports!(installer)
    rescue => e
      Pod::UI.warn e
    end
  end
  config = use_native_modules!

  # Flags change depending on the env values.
  flags = get_default_flags()

  use_react_native!(
    :path => config[:reactNativePath],
    :production => production,
    # to disable hermes on iOS, change `true` to `flags[:hermes_enabled]` and then install pods
    :hermes_enabled => true,
    :fabric_enabled => false, # disable fabric due to incompatiblility with firebase https://rnfirebase.io/
    # disable flipper due to incompatiblility with firebase https://rnfirebase.io/
    # :flipper_configuration => FlipperConfiguration.enabled(["Debug"], { 'Flipper' => '0.157.0' }),
    # An absolute path to your application root.
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )

  pod 'RNVectorIcons', :path => '../node_modules/react-native-vector-icons'

  post_install do |installer|
    react_native_post_install(installer)
    #__apply_Xcode_12_5_M1_post_install_workaround(installer)
    File.delete("./.xcode.env.local") if File.exist?("./.xcode.env.local")
  end
end

```

**Comment by mikehardy** [comment]
>  and got constant crash at startup.

But no crash trace provided? :thinking: 

https://stackoverflow.com/help/how-to-ask

**Comment by CoderCharlie** [comment]
Thanks for your reply.

I don't have the full stack right now as the crash disappeared after I downgraded firebase to 14, expo to 46, react-native to 0.69.6 and reenabled flipper. Here is fragment of error I observed before.
`#6 0x0000000110a3038e in facebook::hermes::inspector::Inspector::installLogHandler() at /.../node_modules/react-native/ReactCommon/hermes/inspector/Inspector.cpp:227`

The crash didn't come from react-native-firebase and directly come from react-native. However, some configurations I made for react-native-firebase to work might related to the crashes, such as disabling flipper, using static link and framework and so on. Therefore, it would be nice to have a validated compatibility versions with react-native and react-native-firebase as well as detailed configurations in Podfile. 

Something similar to this: https://github.com/realm/realm-js/blob/master/COMPATIBILITY.md

**Reply by mikehardy** [reply]
I agree those are nice but they also go out of date quickly and it's difficult to keep up. I provide fully tested working demo here https://github.com/mikehardy/rnfbdemo/blob/main/make-demo.sh and every single commit there worked at the time which gives at least snapshots of compatibility. And "current stable" always works (that is, react-native 0.70.6 + react-native-firebase 16.4.6 right now) with the Podfile created from that project

**Reply by CoderCharlie** [reply]
Thanks, this is exactly what I am looking for. I believe it is also very valuable for all other beginners to have this included on the front page of the documentation.
